CRM/Contact/Form/Task/EmailCommon.php

```diff
@@ -89,7 +89,11 @@ public static function preProcessFromAddress(&$form, $bounce = TRUE) {
     }
 
     $form->_emails = $fromEmailValues;
+    $defaults = array();
     $form->_fromEmails = $fromEmailValues;
+    if (!Civi::settings()->get('allow_mail_from_logged_in_contact')) {
+      $defaults['from_email_address'] = key(CRM_Core_OptionGroup::values('from_email_address', TRUE, NULL, NULL, ' AND is_default = 1'));
```

Comment:
Ok, we shouldn't be calling CRM_Core_OptionGroup::values directly here.  I think the correct function to use would be `CRM_Core_BAO_Domain::getNameAndEmail()` from here: https://github.com/civicrm/civicrm-core/blob/master/CRM/Core/BAO/Domain.php#L172

Reply:
Thanks @mattwire Updated the PR with the above function replacement.